lib/filters/filter_gun_write.js

```diff
@@ -30,26 +36,21 @@ util.inherits(FilterGunWrite, base_filter.BaseFilter);
 FilterGunWrite.prototype.start = function(callback) {
   
   this.gunfire = function(key, data, callback) {
-   if (!this.field && !this.write) { callback(true);return;  }
+   if (!this.field && !this.write) { callback(true);return; }
    else {
     try {
 	if (this.write && this.source) {
-		if(this.source.indexOf('.') !== -1) {
-			var extract = this.source.split('.').reduce(function(a, b) {
-			  return a[b];
-			}, data);
-		} else {
-			var extract = data[this.source];
-		}
-    		if (!extract || !this.source) { 
-			callback('no data! skip'); 
+		var extract = {}; extract[this.source] = Extract(this.source,data);
+    		if (!extract || !this.source) {
+			callback('no data! skip');
 		} else if (this.ttl && this.ttl != 0){
-			gun.get(key).get(this.source).put(extract).later(function(data, key){
-  				this.get('session').put(null);
+			this.gun.get(key).put(extract).later(function(data, key){
+				logger.info('Expiring key:',key);
+  				this.get(key).put(null);
```

Comment:
This looks like it might be a root node, in which putting `null` won't work. (In a graph database, there are nodes, in nodes, there are key/value pairs. Key/value pairs can point to any other node, and key/value pairs themselves can be `null`ed out. However a node itself cannot be `null`. You can maybe fix this by instantiating gun with a root application node, `var gun = Gun(...).get('paStash')`. Let me know if you need help. And let me know how `later` works, if it needs any improvement!

Reply:
After taking a look at the returned set, I ended up using ```this.off();``` to remove the expired item... seems to work as intended! :)